pkg/restapi/vc/operation/operations.go

```diff
@@ -8,94 +8,143 @@ package operation
 type Handler interface {
 	Path() string
 	Method() string
 	Handle() http.HandlerFunc
 }
 
-// New returns CreateCrendential instance
-func New() *Operation {
-	svc := &Operation{}
+// New returns CreateCredential instance
+func New(provider storage.Provider) (*Operation, error) {
+	store, err := provider.OpenStore(credential)
+	if err != nil {
+		return nil, err
+	}
+
+	profileStore := NewProfile(store)
+	svc := &Operation{
+		profileStore: profileStore,
+	}
 	svc.registerHandler()
 
-	return svc
+	return svc, nil
 }
 
-// Operation defines handlers for VC service
+// Operation defines handlers for Edge service
 type Operation struct {
-	handlers []Handler
+	handlers     []Handler
+	profileStore *Profile
 }
 
 func (c *Operation) createCredentialHandler(rw http.ResponseWriter, req *http.Request) {
 	data := CreateCrendential{}
 	err := json.NewDecoder(req.Body).Decode(&data)
 
 	if err != nil {
-		rw.WriteHeader(http.StatusBadRequest)
-		_, err = rw.Write([]byte(fmt.Sprintf("Receieved invalid request: %s", err.Error())))
-
-		if err != nil {
-			log.Errorf("Failed to write response for credential creation failure (unable to read request): %s", err.Error())
-		}
+		c.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for invalid request received")
 
 		return
 	}
 
 	validCredential, err := createCredential(&data)
 
 	if err != nil {
-		rw.WriteHeader(http.StatusBadRequest)
-		_, err = rw.Write([]byte(fmt.Sprintf("validation failed for vc: %s", err.Error())))
-
-		if err != nil {
-			log.Errorf("Failed to write response for create credential failure: %s", err.Error())
-		}
+		c.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for create credential failure")
 
 		return
 	}
 
+	rw.WriteHeader(http.StatusCreated)
 	c.writeResponse(rw, validCredential)
 }
 
-// writeResponse writes interface value to response
-func (c *Operation) writeResponse(rw http.ResponseWriter, v interface{}) {
+func (c *Operation) createProfileHandler(rw http.ResponseWriter, req *http.Request) {
+	data := ProfileRequest{}
+	err := json.NewDecoder(req.Body).Decode(&data)
+
+	if err != nil {
+		c.writeErrorResponse(rw, http.StatusBadRequest, "Failed to write response for invalid request received")
+
+		return
+	}
+
+	profileResponse, err := c.createProfile(&data)
+
+	if err != nil {
+		c.writeErrorResponse(rw, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
 	rw.WriteHeader(http.StatusCreated)
-	err := json.NewEncoder(rw).Encode(v)
-	// as of now, just log errors for writing response
+	c.writeResponse(rw, profileResponse)
+}
+
+func (c *Operation) getProfileHandler(rw http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	profileID := vars[profilePathVariable]
+
+	profileResponseJSON, err := c.profileStore.GetProfile(profileID)
 	if err != nil {
-		log.Errorf("Unable to send error response, %s", err)
+		if err == errProfileNotFound {
+			rw.WriteHeader(http.StatusNotFound)
```

Comment:
header is also written in write error response, no need to write twice

Reply:
done